### PR TITLE
Migrate OV to 2022.1

### PR DESCRIPTION
**Pull request by kthui**

### CODE REVIEW

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

**Comment by kthui** [comment]
Added patch into setting batch size: 
```
$ git diff 7dd68dcf124ec43f2ad1e3ede3d8f88a59c3f6ef 5c5c64728635c80e0d2fe96d54ae48e4ae0097ce
diff --git a/src/openvino.cc b/src/openvino.cc
index e2dcc28..0b6ee27 100644
--- a/src/openvino.cc
+++ b/src/openvino.cc
@@ -552,8 +552,10 @@ ModelState::ValidateInputs(const size_t expected_input_cnt)
       ov_model_, ppp.build(), "apply model input preprocessing");
 
   // Configuring the model to handle the max_batch_size
-  RETURN_IF_OPENVINO_ERROR(
-      ov::set_batch(ov_model_, MaxBatchSize()), "setting max batch size");
+  if (MaxBatchSize()) {
+    RETURN_IF_OPENVINO_ERROR(
+        ov::set_batch(ov_model_, MaxBatchSize()), "setting max batch size");
+  }
 
   return nullptr;  // success
 }
```
This change is force updated into the "Upgrade to api 2.0" commit.
Will use new commits moving forward.

File: README.md [hunk 1]
```diff
@@ -78,10 +78,10 @@ but the listed CMake argument can be used to override.
...
-* `CPU_THREADS_NUM`: Number of threads to use for inference on the CPU. Should be a non-negative number.
-* `ENFORCE_BF16`: Enforcing of floating point operations execution in bfloat16 precision on platforms with native bfloat16 support. Possible values are `YES` or `NO`.
-* `CPU_BIND_THREAD`: Enable threads->cores (`YES`, default), threads->(NUMA)nodes (`NUMA`) or completely disable (`NO`) CPU threads pinning for CPU-involved inference.
-* `CPU_THROUGHPUT_STREAMS`: Number of streams to use for inference on the CPU. Default value is determined automatically for a device. Please note that although the automatic selection usually provides a reasonable performance, it still may be non-optimal for some cases, especially for very small networks. Also, using nstreams>1 is inherently throughput-oriented option, while for the best-latency estimations the number of streams should be set to 1.
+* `INFERENCE_NUM_THREADS`: Number of threads to use for inference on the CPU. Should be a non-negative number.
+* `COMPILATION_NUM_THREADS`: Number of threads to use for model compilation. Should be a non-negative number.
```
[the remark this answers is not in the document]
Comment (kthui) [reply README.md:82]: It sets the number of threads for all compilation tasks.
I intent it to be used for the [loading step](https://docs.openvino.ai/latest/openvino_2_0_inference_pipeline.html#load-the-model-to-the-device). It used to be called `core.LoadNetwork`, and in API 2.0 it is called `core.compile_model`.
It is a bit confusing, maybe I should describe it as "Number of threads to use for model compilation/loading."?

File: src/openvino.cc [hunk 2]
```diff
@@ -1118,12 +1080,14 @@ ModelInstanceState::SetInputTensors(
       }
 
       // Set the input blob to the buffer without allocating any new memory
-      auto data_blob = WrapInputBufferToBlob(
-          input_tensor_infos[input_name]->getTensorDesc(), input_buffer,
-          batchn_byte_size);
+      ov::Tensor expected_tensor =
```
[the remark this answers is not in the document]
Comment (kthui) [reply src/openvino.cc:1085]: Yes. Changed to `temp_input_tensor`.

File: src/openvino.cc [hunk 3]
```diff
@@ -1118,12 +1080,14 @@ ModelInstanceState::SetInputTensors(
...
+          infer_request_.get_tensor(std::string(input_name));
```
[the remark this answers is not in the document]
Comment (kthui) [reply src/openvino.cc:1086]: Yes. Updated to read directly.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanmayv25) [reply README.md]: You can document like:
"Maximum number of threads that can be used for compilation tasks. Should be a non-negative number."
This is how it has been documented here: https://docs.openvino.ai/latest/groupov_runtime_cpp_prop_api.html#doxid-group-ov-runtime-cpp-prop-api-1ga91555d2fad22aa802aa9d36698805755

File: README.md [hunk 4]
```diff
@@ -78,10 +78,10 @@ but the listed CMake argument can be used to override.
...
-* `CPU_THREADS_NUM`: Number of threads to use for inference on the CPU. Should be a non-negative number.
-* `ENFORCE_BF16`: Enforcing of floating point operations execution in bfloat16 precision on platforms with native bfloat16 support. Possible values are `YES` or `NO`.
-* `CPU_BIND_THREAD`: Enable threads->cores (`YES`, default), threads->(NUMA)nodes (`NUMA`) or completely disable (`NO`) CPU threads pinning for CPU-involved inference.
-* `CPU_THROUGHPUT_STREAMS`: Number of streams to use for inference on the CPU. Default value is determined automatically for a device. Please note that although the automatic selection usually provides a reasonable performance, it still may be non-optimal for some cases, especially for very small networks. Also, using nstreams>1 is inherently throughput-oriented option, while for the best-latency estimations the number of streams should be set to 1.
+* `INFERENCE_NUM_THREADS`: Number of threads to use for inference on the CPU. Should be a non-negative number.
+* `COMPILATION_NUM_THREADS`: Number of threads to use for model loading/compilation. Should be a non-negative number.
```
Comment (tanmayv25) [inline README.md:82]: Maximum number of threads that can be used for compilation tasks.
https://docs.openvino.ai/latest/groupov_runtime_cpp_prop_api.html#doxid-group-ov-runtime-cpp-prop-api-1ga91555d2fad22aa802aa9d36698805755

File: README.md [hunk 5]
```diff
@@ -78,10 +78,10 @@ but the listed CMake argument can be used to override.
...
 
 * `CPU_EXTENSION_PATH`: Required for CPU custom layers. Absolute path to a shared library with the kernels implementations.
-* `CPU_THREADS_NUM`: Number of threads to use for inference on the CPU. Should be a non-negative number.
-* `ENFORCE_BF16`: Enforcing of floating point operations execution in bfloat16 precision on platforms with native bfloat16 support. Possible values are `YES` or `NO`.
-* `CPU_BIND_THREAD`: Enable threads->cores (`YES`, default), threads->(NUMA)nodes (`NUMA`) or completely disable (`NO`) CPU threads pinning for CPU-involved inference.
-* `CPU_THROUGHPUT_STREAMS`: Number of streams to use for inference on the CPU. Default value is determined automatically for a device. Please note that although the automatic selection usually provides a reasonable performance, it still may be non-optimal for some cases, especially for very small networks. Also, using nstreams>1 is inherently throughput-oriented option, while for the best-latency estimations the number of streams should be set to 1.
+* `INFERENCE_NUM_THREADS`: Number of threads to use for inference on the CPU. Should be a non-negative number.
```
Comment (tanmayv25) [inline README.md:81]: Maximum number of threads that can be used for inference tasks.

File: README.md [hunk 6]
```diff
@@ -78,10 +78,10 @@ but the listed CMake argument can be used to override.
...
+* `COMPILATION_NUM_THREADS`: Number of threads to use for model loading/compilation. Should be a non-negative number.
+* `HINT_BF16`: Hint of floating point operations execution in bfloat16 precision on platforms with native bfloat16 support. Possible value is `YES`.
+* `NUM_STREAMS`: Enable threads->cores (`YES`, default) or threads->(NUMA)nodes (`NUMA`) CPU threads pinning for CPU-involved inference.
```
Comment (tanmayv25) [inline README.md:84]: The number of executor logical partitions. Enable ....

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

**Comment by tanmayv25** [comment]
Change the documentation language to match more closely with intel's description of the parameters.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kthui) [reply README.md]: Yes. Updated.

File: README.md (hunk 4 above)
Comment (kthui) [reply README.md]: Yes. Updated.

File: README.md (hunk 5 above)
Comment (kthui) [reply README.md]: Yes. Updated.

File: README.md (hunk 6 above)
Comment (kthui) [reply README.md]: Yes. Updated.

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**